### ISSUE TITLE
goimport AST walking...

### ISSUE THREAD

**Issue by sean-** [kardianos/govendor#107]
When trying to use `govendor` with a project, it looks like its getting hung up traversing imports that only work on, for example, appengine, and won't ever work for non-appengine environments.  For example:

```go
package mysql

import (
	"appengine/cloudsql"
)
```

from https://github.com/go-sql-driver/mysql/blob/master/appengine.go#L14

The dependency graph from `govendor list -v`:

```
 v  github.com/go-sql-driver/mysql ::/vendor/github.com/go-sql-driver/mysql
    ├── github.com/hashicorp/vault/builtin/logical/mysql
    └── github.com/hashicorp/vault/physical
[snip]
  m appengine/cloudsql
    └── github.com/hashicorp/vault/vendor/github.com/go-sql-driver/mysql
```

I'm not sure what the right fix is for this.  At first pass it appears as though `+build` flag needs to be taken in to consideration when walking the AST, but that appears to only be a necessary step but is by itself an incomplete fix.  Initially I was hoping the problem was as simple as ignoring files tagged with `appengine`:

```
diff --git a/context/resolve.go b/context/resolve.go
index 28cd596..82a223c 100644
--- a/context/resolve.go
+++ b/context/resolve.go
@@ -156,7 +156,7 @@ func (ctx *Context) addFileImports(pathname, gopath string) error {
 	// if file has "// +build ignore", can mix package main with normal package.
 	isIgnore := false
 	for _, tag := range tags {
-		if tag == "ignore" {
+		if tag == "ignore" || tag == "appengine" {
 			isIgnore = true
 		}
 	}
```

but there is still an orphaned dependency on `appengine/cloudsql` that needs to be removed from the context after `addFileImports()` has finished:

```
  m appengine/cloudsql
```

I haven't had time to chase this further, but after running through `addFileImports` anything with no dependencies or references should be removed from the context.  ?

**Comment by kardianos** [comment]
I've added a note about this in the [FAQ](https://github.com/kardianos/govendor/blob/master/doc/faq.md#q-im-getting-missing-packages-from-appengine-but-i-dont-care-about-appengine-how-do-i-ignore-these-pacakges) . Could you see if this answers your question?

**Comment by sean-** [comment]
It does, thank you!  This did unblock other problems with `fetch` and its cache directory that I'm going to post as a new issue.
